### PR TITLE
Update centrality definition for FXT

### PR DESCRIPTION
**Pull request by nigmatkulov**
Update of the centrality definition for FXT (3, 3.2, 3.5, 3.9, and 7.7 GeV) and Collider (7.7 and 200 GeV)

### CODE REVIEW

**Comment by plexoos** [comment]
I'm not the right person to review this but let me know and I can approve. AFAICT, the changes will not affect reco jobs.

**Comment by nigmatkulov** [comment]
Thank you very much. I'll also ask our new code maintainer for the StRefMultCorr to look at it.

File: StRoot/StRefMultCorr/BadRun.h [hunk 1]
```diff
@@ -66,16 +69,47 @@ const Int_t badrun_grefmult_2016[nBadRun_grefmult_2016] = {
...
+// 1. Au+Au 31.2 AGeV (3) = 7.7 GeV
+// 2. Au+Au 7.3 AGeV (4) = 3.9 GeV
+// 3. Au+Au 5.75 AGeV (8) = 3.5 GeV
+//
+const Int_t nBadRun_refmult_2020 = 15;
+const Int_t badrun_refmult_2020[nBadRun_refmult_2020] = {
+  // Au+Au 31.2 AGeV (7.7 GeV)
+  21029002, 21029013, 21029027,
+  // Au+Au 7.3 AGeV (3.9 GeV)
+  21035008, 21035011, 21035012, 21035014,
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/BadRun.h]: Typo: Bad run should be 21036012 in stead of 21035012

```suggestion
  21035008, 21035011, 21036012, 21035014,
```

File: StRoot/StRefMultCorr/BadRun.h [hunk 2]
```diff
@@ -66,16 +69,47 @@ const Int_t badrun_grefmult_2016[nBadRun_grefmult_2016] = {
...
+  // Au+Au 5.75 AGeV (3.5 GeV)
+  355020, 20355021, 21044023, 21045024, 21045025, 21044027, 21044035, 21045004
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/BadRun.h]: Typo: 355020 -> 20355020
```suggestion
  20355020, 20355021, 21044023, 21045024, 21045025, 21044027, 21044035, 21045004
```

File: StRoot/StRefMultCorr/BadRun.h [hunk 3]
```diff
@@ -66,16 +69,47 @@ const Int_t badrun_grefmult_2016[nBadRun_grefmult_2016] = {
...
   // Au+Au 200 GeV
-  20191005, 20191015, 20192001, 20193001, 20193019
+  20191005, 20191015, 20192001, 20193001, 20193019,
+  // Au+Au 4.59 AGeV (3.2 GeV)
+  20180005, 20180006, 20180019, 20180025, 20181016, 20182034, 20183001, 20183013, 20183014, 20183019,
+  // Au+Au 7.3 AGeV (1) = 3.9 GeV
+  20169051
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/BadRun.h]: Taken from [official FXT bad run list](https://drupal.star.bnl.gov/STAR/pwg/common/bes-ii-run-qa/FXT-datasets)

```suggestion
  20107029, 20113042, 20113043, 20169033, 20169043
```

File: StRoot/StRefMultCorr/Param.h [hunk 4]
```diff
@@ -481,10 +471,96 @@ const string mParamStr_ref1[nID_ref1][nSet_ref1] = {
...
+    "2020:7.3:21028011,21029036:198,202",         // Year, energy, run start end, Vz range
+    "10,15,21,28,38,49,63,79,98,119,142,169,200,233,271,315",  // Multiplicity values for the centrality classes
+    "160",                        // Normalization start (of Glauber to the data)
+    "0.,0.,0.,0.,0.,0.,0.,0.",    // Vz correction parameters (switched to new scheme, parameters defined near the shape corrections)
+    "9.63078e-1,6.54124e-02,4.68589e-02,-2.43086e-02,8.15939e-03,0,4.50613e-02,-1.14802e-03",   // Trigger efficiency
+    "0.,0."                       // Luminosity correction
+  }
+};
+
+
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:563]: Second run ID range of 5.75 GeV Run 20
```suggestion
  // mParameterIndex = 6
  // Run 20 Au+Au 5.75 GeV (sqrt(s_NN)=3.5 GeV), Trigger ID = 720000
  {
    "2020:5.75:21044023,21045011:198,202",         // Year, energy, run start end, Vz range
    "6,8,12,16,21,28,37,47,59,73,89,108,128,153,181,216",  // Multiplicity values for the centrality classes
    "107",                        // Normalization start (of Glauber to the data)
    "0.,0.,0.,0.,0.,0.,0.,0.",   // Vz correction parameters (switched to new scheme, parameters defined near the shape corrections)
    "1.22,-0.137,0.0532,0.00845,-0.0690,0,0.1,0.00546",   // Trigger efficiency
    "0.,0."                      // Luminosity correction
  },
  
```

File: StRoot/StRefMultCorr/Param.h [hunk 5]
```diff
@@ -481,10 +471,96 @@ const string mParamStr_ref1[nID_ref1][nSet_ref1] = {
...
+    "0,0" // Luminosity correction
   }
 };
 
+//=======================================================================================
+//if you want to add new parameters, please always add after the current parameters, do not insert between them
+//the order of them is important!!!!!!!!!!!!!
+// FXT section
+//=======================================================================================
+const int nID_ref5  = 6;//after add N more new group of parameters, need to update nID_ref1 += N
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:493]: Run ID of 5.75 GeV Run 20 dataset overlaps other Run 20 FXT datasets. To separate 5.75 GeV Run 20 using two mParameterIndex.
```suggestion
const int nID_ref5  = 7;//after add N more new group of parameters, need to update nID_ref1 += N
```

File: StRoot/StRefMultCorr/StRefMultCorr.cxx [hunk 6]
```diff
@@ -160,153 +172,340 @@ Bool_t StRefMultCorr::passnTofMatchRefmultCut(Double_t refmult, Double_t ntofmat
...
+      b0=18.0459;
+      b1=1.32913;
+      b2=-0.000929385;
+      b3=1.53176e-06;
+      b4=-9.911e-10;
+      c0=-18.7481;
+      c1=0.785467;
+      c2=2.12757e-05;
+      c3=3.4805e-07;
+      c4=-3.80776e-100;
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/StRefMultCorr.cxx]: Typo
```suggestion
      c4=-3.80776e-10;
```

File: StRoot/StRefMultCorr/StRefMultCorr.cxx [hunk 7]
```diff
@@ -160,153 +172,340 @@ Bool_t StRefMultCorr::passnTofMatchRefmultCut(Double_t refmult, Double_t ntofmat
...
+      c0=-24.3335976220474;
+      c1=1.93207052914575;
+      c2=0.0042539477677528;
+      c3=0.000725893349545147;
+      c4=-6.29726910263091e-06;
+
+      refmultcutmax = calcPileUpRefMult(ntofmatch, b0, b1, b2, b3, b4);
+      refmultcutmin = calcPileUpRefMult(ntofmatch, c0, c1, c2, c3, c4);
+      notPileUp = isInPileUpRefMultLimits(refmult, refmultcutmin, refmultcutmax);  
+    }
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/StRefMultCorr.cxx:494]: Duplication for the second run ID range of 5.75 GeV Run 20
```suggestion
    }
    else if (mParameterIndex == 6) { // Run 20 Au+Au 5.75 GeV (sqrt(s_NN)=3.5 GeV)
      b0=23.28;
      b1=5.247;
      b2=0.04037;
      b3=-1.206e-3;
      b4=5.792e-6;
      c0=-14.82;
      c1=1.583;
      c2=0.02684;
      c3=4.605e-5;
      c4=-2.410e-6;

      refmultcutmax = calcPileUpRefMult(ntofmatch, b0, b1, b2, b3, b4);
      refmultcutmin = calcPileUpRefMult(ntofmatch, c0, c1, c2, c3, c4);
      notPileUp = isInPileUpRefMultLimits(refmult, refmultcutmin, refmultcutmax);  
    }
```

**Review by lbavinh — CHANGES_REQUESTED** [review]
Some minor changes need to be committed before merging

File: StRoot/StRefMultCorr/BadRun.h [hunk 8]
```diff
@@ -66,16 +69,47 @@ const Int_t badrun_grefmult_2016[nBadRun_grefmult_2016] = {
 // 1. Au+Au 19.6 GeV (374)
 // 2. Au+Au 14.6 GeV (bad runs: 153 + injection runs: 403 = 556)
 // 3. Au+Au 200 GeV (5)
+// 4. Au+Au 4.59 AGeV (10) = 3.2 GeV
+// 5. Au+Au 7.3 AGeV (1) = 3.9 GeV
 //
-const Int_t nBadRun_refmult_2019 = 935;
+const Int_t nBadRun_refmult_2019 = 946;
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/BadRun.h]: Add bad runs from Au+Au 7.3 AGeV = 3.9 GeV 
```suggestion
const Int_t nBadRun_refmult_2019 = 951;
```

File: StRoot/StRefMultCorr/BadRun.h (hunk 3 above)
Comment (nigmatkulov) [reply StRoot/StRefMultCorr/BadRun.h]: Did you ask if Erik used those bad runs for the 3.9 GeV energy?

Comment (lbavinh) [reply StRoot/StRefMultCorr/BadRun.h]: Erik did use these official bad runs from the QA board.
The 20169051 run isn't in QA board list, but it is in [Erik's presentation](https://drupal.star.bnl.gov/STAR/blog/eloyd/Run-19-39GeVFXT-Centrality-Definition).

Comment (lbavinh) [reply StRoot/StRefMultCorr/BadRun.h]: Then, maybe we should also exclude 20169051
```suggestion
  20107029, 20113042, 20113043, 20169033, 20169043, 20169051
```

File: StRoot/StRefMultCorr/Param.h (hunk 5 above)
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:493]:
```suggestion
const int nID_ref5  = 7;//after add N more new group of parameters, need to update nID_ref1 += N
```

File: StRoot/StRefMultCorr/Param.h [hunk 9]
```diff
@@ -481,10 +471,96 @@ const string mParamStr_ref1[nID_ref1][nSet_ref1] = {
...
+    "118",                        // Normalization start (of Glauber to the data)
+    "0.,0.,0.,0.,0.,0.,0.,0.",   // Vz correction parameters (switched to new scheme, parameters defined near the shape corrections)
+    "1.2680,-0.13,0.0593,0.0179,-0.0879,0,0.120,0.00714",   // Trigger efficiency
+    "0.,0."                      // Luminosity correction
+  },
+
+  // mParameterIndex = 2
+  // Run 20 Au+Au 5.75 GeV (sqrt(s_NN)=3.5 GeV), Trigger ID = 720000
+  {
+    "2020:5.75:20355020,21045011:198,202",         // Year, energy, run start end, Vz range
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:521]:
```suggestion
    "2020:5.75:20355020,20355021:198,202",         // Year, energy, run start end, Vz range
```

File: StRoot/StRefMultCorr/Param.h [hunk 10]
```diff
@@ -481,10 +471,96 @@ const string mParamStr_ref1[nID_ref1][nSet_ref1] = {
...
+    "141",                        // Normalization start (of Glauber to the data)
+    "0.,0.,0.,0.,0.,0.,0.,0.",    // Vz correction parameters (switched to new scheme, parameters defined near the shape corrections)
+    "1.017,0.0609,0.0744,-0.106,-0.003,0,0.0824,-3.91e-4",   // Trigger efficiency
+    "0.,0."                       // Luminosity correction
+  },
+
+  // mParameterIndex = 5
+  // Run 20 Au+Au 31.2 GeV (sqrt(s_NN)=7.7 GeV), Trigger ID = 770000
+  {
+    "2020:7.3:21028011,21029036:198,202",         // Year, energy, run start end, Vz range
```
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:554]: Typo
```suggestion
    "2020:31.2:21028011,21029036:198,202",         // Year, energy, run start end, Vz range
```

File: StRoot/StRefMultCorr/Param.h (hunk 4 above)
Comment (lbavinh) [inline StRoot/StRefMultCorr/Param.h:563]:
```suggestion
  // mParameterIndex = 6
  // Run 20 Au+Au 5.75 GeV (sqrt(s_NN)=3.5 GeV), Trigger ID = 720000
  {
    "2020:5.75:21044023,21045011:198,202",         // Year, energy, run start end, Vz range
    "6,8,12,16,21,28,37,47,59,73,89,108,128,153,181,216",  // Multiplicity values for the centrality classes
    "107",                        // Normalization start (of Glauber to the data)
    "0.,0.,0.,0.,0.,0.,0.,0.",   // Vz correction parameters (switched to new scheme, parameters defined near the shape corrections)
    "1.22,-0.137,0.0532,0.00845,-0.0690,0,0.1,0.00546",   // Trigger efficiency
    "0.,0."                      // Luminosity correction
  },
  
```

**Review by lbavinh — CHANGES_REQUESTED** [review]
Split 5.75 GeV Run 20 into two mParameterIndex due to long data taking.

**Review by lbavinh — APPROVED** [review]

**Comment by nigmatkulov** [comment]
Current status: all ready with only non-deformed Glauber 3.2, 3.5, and 3.9 GeV. Next step is to update the values for deformed Glauber.

**Review by zsweger — APPROVED** [review]

**Pull request merged**